cpp/src/prims/per_v_transform_reduce_incoming_outgoing_e.cuh

```diff
@@ -513,24 +513,20 @@ void per_v_transform_reduce_e(raft::handle_t const& handle,
 
   static_assert(is_arithmetic_or_thrust_tuple_of_arithmetic<T>::value);
 
-  [[maybe_unused]] std::conditional_t<GraphViewType::is_storage_transposed,
-                                      edge_src_property_t<GraphViewType, T>,
-                                      edge_dst_property_t<GraphViewType, T>>
-    minor_tmp_buffer(handle);  // relevant only when (GraphViewType::is_multi_gpu && !update_major
+  using minor_tmp_buffer_type = std::conditional_t<GraphViewType::is_storage_transposed,
+                                                   edge_src_property_t<GraphViewType, T>,
+                                                   edge_dst_property_t<GraphViewType, T>>;
+  std::unique_ptr<minor_tmp_buffer_type> minor_tmp_buffer{};
```

Comment:
Would an explicit `std::optional<minor_tmp_buffer_type>` defined here and initialized to `std::nullopt` also work?  That's a code pattern we use quite a bit.  Would that perhaps be a closer idiom to `[[maybe_unused]]` until the warning issue is corrected in the compiler?

Approving either way, just wondering.

Reply:
It may work as well but std::optional is for run-time resolution. Here whether `minor_tmp_buffer` is used or not can be determined in compile-time.

Reply:
Yeah... but your comment reminded me that I didn't add [[maybe_unused]] in this update. Adding this to `minor_tmp_buffer` declaration.